Testscripts/Windows/HB-CLOCKSYNC.ps1

```diff
@@ -0,0 +1,305 @@
+		$storageType = 'StandardSSD_LRS'
+		$dataDiskName = $vmName + '_datadisk1'
+
+		#region Generate constants.sh
+		# We need to add extra parameters to constants.sh file apart from parameter properties defined in XML.
+		# Hence, we are generating constants.sh file again in test script.
+
+		Write-LogInfo "Generating constants.sh ..."
+		$constantsFile = "$LogDir\constants.sh"
+		foreach ($TestParam in $CurrentTestData.TestParameters.param) {
+			Add-Content -Value "$TestParam" -Path $constantsFile
+			Write-LogInfo "$TestParam added to constants.sh"
+		}
+
+		Write-LogInfo "constants.sh created successfully..."
+		#endregion
+
+		#region Add a new swap disk to Azure VM
+		$diskConfig = New-AzDiskConfig -SkuName $storageType -Location $location -CreateOption Empty -DiskSizeGB 1024
+		$dataDisk1 = New-AzDisk -DiskName $dataDiskName -Disk $diskConfig -ResourceGroupName $rgName
+
+		$vm = Get-AzVM -Name $vmName -ResourceGroupName $rgName
+		Start-Sleep -seconds 30
+		$vm = Add-AzVMDataDisk -VM $vm -Name $dataDiskName -CreateOption Attach -ManagedDiskId $dataDisk1.Id -Lun 1
+		Start-Sleep -seconds 30
+
+		$ret_val = Update-AzVM -VM $vm -ResourceGroupName $rgName
+		Write-LogInfo "Updated the VM with a new data disk"
+		Write-LogInfo "Waiting for 30 seconds for configuration sync"
+		# Wait for disk sync with Azure host
+		Start-Sleep -seconds 30
+
+		# Verify the new data disk addition
+		if ($ret_val.IsSuccessStatusCode) {
+			Write-LogInfo "Successfully add a new disk to the Resource Group, $($rgName)"
+		} else {
+			Write-LogErr "Failed to add a new disk to the Resource Group, $($rgname)"
+			throw "Failed to add a new disk"
+		}
+
+		# TODO: This will be revised with other distro. hwclock output format might be different.
+		$getyear = @"
+source utils.sh
+hwclock > _time
+if [[ `$DISTRO_NAME == 'ubuntu' && `$DISTRO_VERSION == '16.04' ]]; then
+	cat _time | cut -d ' ' -f 4 > timestamp.log
+else
+	cat _time | cut -d '-' -f 1 > timestamp.log
+fi
+"@
+		Set-Content "$LogDir\getyear.sh" $getyear
+
+		$testcommand = @"
+echo disk > /sys/power/state
+"@
+		Set-Content "$LogDir\test.sh" $testcommand
+
+		#region Upload files to VM
+		foreach ($VMData in $AllVMData) {
+			Copy-RemoteFiles -uploadTo $VMData.PublicIP -port $VMData.SSHPort -files "$constantsFile,$($CurrentTestData.files),$LogDir\*.sh" -username $user -password $password -upload
+			Write-LogInfo "Copied the script files to the VM"
+		}
+		#endregion
+
+		# Configuration for the hibernation
+		Run-LinuxCmd -ip $AllVMData.PublicIP -port $AllVMData.SSHPort -username $user -password $password -command "./SetupHbKernel.sh" -RunInBackground -runAsSudo -ignoreLinuxExitCode:$true | Out-Null
+		Write-LogInfo "Executed SetupHbKernel script inside VM"
+
+		# Wait for kernel compilation completion. 90 min timeout
+		$timeout = New-Timespan -Minutes $maxKernelCompileMin
+		$sw = [diagnostics.stopwatch]::StartNew()
+		while ($sw.elapsed -lt $timeout){
+			$vmCount = $AllVMData.Count
```

Comment:
ditto in another PR

Reply:
Fixed in e3b25b3b590e7e0d5575dc6cb1e39e2289eaa1db